frontend/src/components/CreateStudentModal.tsx

```diff
@@ -0,0 +1,46 @@
+import { ErrorOutline } from '@mui/icons-material';
+import { Modal, ModalDialog, Snackbar, Typography } from '@mui/joy';
+import { useState } from 'react';
+import { StudentRegisterForm } from './StudentRegisterForm.tsx';
+
+export function CreateStudentModal({ isOpen, setOpen }: { isOpen: boolean; setOpen: (open: boolean) => void }) {
+  const handleModalClose = () => setOpen(false);
+  const [showSnackbar, setShowSnackbar] = useState(false);
+  const [showSnackbarError, setShowSnackbarError] = useState(false);
```

Comment:
```suggestion
  const [showErrorSnackbar, setShowErrorSnackbar] = useState(false);
```
you don't show an Error because snackbar is not working you show an error in snackbar (i know Kiearan did it this way but i didn't see it then).